io/plugins/eu.esdihumboldt.hale.io.json.ui/src/eu/esdihumboldt/hale/io/json/ui/TypeSelectionPage.java

```diff
@@ -0,0 +1,272 @@
+import org.eclipse.swt.widgets.Button;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Group;
+import org.eclipse.swt.widgets.Label;
+
+import eu.esdihumboldt.hale.common.core.io.IOProvider;
+import eu.esdihumboldt.hale.common.core.io.Value;
+import eu.esdihumboldt.hale.common.instance.io.InstanceReader;
+import eu.esdihumboldt.hale.common.schema.model.TypeDefinition;
+import eu.esdihumboldt.hale.io.json.JsonInstanceReader;
+import eu.esdihumboldt.hale.io.json.ui.util.EnumJSONAutoOrderTypes;
+import eu.esdihumboldt.hale.ui.HaleWizardPage;
+import eu.esdihumboldt.hale.ui.common.definition.selector.TypeDefinitionSelector;
+import eu.esdihumboldt.hale.ui.io.IOWizardPage;
+import eu.esdihumboldt.hale.ui.io.config.AbstractConfigurationPage;
+import eu.esdihumboldt.hale.ui.io.instance.InstanceReaderConfigurationPage;
+import eu.esdihumboldt.hale.ui.util.viewer.EnumContentProvider;
+
+/**
+ * Configuration page for selecting the feature type for GeoJSON/JSON instances.
+ * 
+ * @author Emanuela Epure
+ */
+@SuppressWarnings("restriction")
+public class TypeSelectionPage extends InstanceReaderConfigurationPage {
+
+	private Composite page;
+	private Button forceUsageOfDefaultSelectedType;
+	private TypeDefinitionSelector selectorFeatureTypes;
+	private ComboViewer readModeOrderCombo;
+	private Label setTypeLabel;
+
+	/**
+	 * default constructor
+	 */
+	public TypeSelectionPage() {
+		super("selectType");
+
+		setTitle("Feature type");
+		setDescription("Select the Feature types matching your data.");
+	}
+
+	/**
+	 * @see AbstractConfigurationPage#enable()
+	 */
+	@Override
+	public void enable() {
+		// nothing
+	}
+
+	/**
+	 * @see AbstractConfigurationPage#disable()
+	 */
+	@Override
+	public void disable() {
+		// do nothing
+	}
+
+	/**
+	 * @see HaleWizardPage#onShowPage(boolean)
+	 */
+	@Override
+	protected void onShowPage(boolean firstShow) {
+		super.onShowPage(firstShow);
+
+		page.setLayout(new GridLayout(2, false));
+		GridData layoutData = new GridData();
+		layoutData.widthHint = 200;
+		
+		if (firstShow) {
+
+			Group autodetectOrderGroup = new Group(page, SWT.NONE);
+			autodetectOrderGroup.setText("Choose the most appropriate read mode:");
+			GridLayoutFactory.swtDefaults().numColumns(1).applyTo(autodetectOrderGroup);
+			GridDataFactory.fillDefaults().grab(true, false).applyTo(autodetectOrderGroup);
+
+			readModeOrderCombo = new ComboViewer(autodetectOrderGroup);
+			GridDataFactory.fillDefaults().grab(true, false).applyTo(readModeOrderCombo.getControl());
+			readModeOrderCombo.setContentProvider(EnumContentProvider.getInstance());
+			readModeOrderCombo.setLabelProvider(new LabelProvider() {
+
+				@Override
+				public String getText(Object element) {
+					if (element instanceof EnumJSONAutoOrderTypes) {
+						return ((EnumJSONAutoOrderTypes) element).getJsonModeOrder();
+					}
+					return super.getText(element);
+				}
+
+			});
+			readModeOrderCombo.setInput(EnumJSONAutoOrderTypes.class);
```

Comment:
I don't see the initial selection being set on the combo. The initial selection should reflect the default for the reader (`auto`).